packit/patches.py

```diff
@@ -496,6 +517,9 @@ def create_patches(
                     patches, commits, destination, files_to_ignore
                 )
                 patch_list = self.process_git_am_style_patches(patch_list)
+                repo = is_git_repo(destination)
+                if repo:
+                    patch_list = self.check_patch_changes(patch_list, repo)
```

Comment:
we should only do this for propose-downstream and make sure all patches are correctly created for `srpm`

Reply:
That's what the `if repo` part is doing. But to be honest, I'm not entierly sure I understand this whole code entierly well, it's super convoluted. I'll check one more time whether this is the right place to do this check of the patches.

Reply:
SRPM builds would definitely start failing if we filtered out all the patches :) so if the current code works for you in the glibc repo (and tests are passing), then the `if` condition should indeed be doing the trick

Reply:
Okay, so I moved undoing the identical patches to `api.sync_release()` after the `create_patches` call, which made checking for the git repo unnecessary.

It seems easier to understand what's happening this way.

Let me know if you think otherwise.